pkg/controller/bastion/actuator.go

```diff
@@ -42,31 +43,15 @@ func newActuator(mgr manager.Manager) bastion.Actuator {
 	}
 }
 
-func getBastionInstance(ctx context.Context, gcpclient gcpclient.Interface, opt *Options) (*compute.Instance, error) {
-	instance, err := gcpclient.Instances().Get(opt.ProjectID, opt.Zone, opt.BastionInstanceName).Context(ctx).Do()
-	if err != nil {
-		if googleError, ok := err.(*googleapi.Error); ok && googleError.Code == http.StatusNotFound {
-			return nil, nil
-		}
-		return nil, err
-	}
-	return instance, nil
-}
-
-func getFirewallRule(ctx context.Context, gcpclient gcpclient.Interface, opt *Options, firewallRuleName string) (*compute.Firewall, error) {
-	firewall, err := gcpclient.Firewalls().Get(opt.ProjectID, firewallRuleName).Context(ctx).Do()
-	if err != nil {
-		if googleError, ok := err.(*googleapi.Error); ok && googleError.Code == http.StatusNotFound {
-			return nil, nil
-		}
-		return nil, err
-	}
-	return firewall, nil
+func getBastionInstance(ctx context.Context, client gcpclient.ComputeClient, opt *Options) (*compute.Instance, error) {
+	instance, err := client.GetInstance(ctx, opt.Zone, opt.BastionInstanceName)
+	return instance, gcpclient.IgnoreNotFoundError(err)
 }
 
-func createFirewallRuleIfNotExist(ctx context.Context, log logr.Logger, gcpclient gcpclient.Interface, opt *Options, firewallRule *compute.Firewall) error {
-	if _, err := gcpclient.Firewalls().Insert(opt.ProjectID, firewallRule).Context(ctx).Do(); err != nil {
-		if googleError, ok := err.(*googleapi.Error); ok && googleError.Code == http.StatusConflict {
+func createFirewallRuleIfNotExist(ctx context.Context, log logr.Logger, client gcpclient.ComputeClient, firewallRule *compute.Firewall) error {
+	if _, err := client.InsertFirewallRule(ctx, firewallRule); err != nil {
+		var googleError *googleapi.Error
+		if errors.As(err, &googleError) && googleError.Code == http.StatusConflict {
```

Comment:
maybe you can use https://github.com/gardener/gardener-extension-provider-gcp/blob/c73f2b87c2b67c9f5a5907113adb47b478342881/pkg/gcp/client/errors.go#L15